src/bidiMapper/modules/context/NavigationTracker.ts

```diff
@@ -400,66 +409,57 @@ export class NavigationTracker {
     }
 
     navigation.isFragmentNavigation = loaderId === undefined;
+  }
 
-    if (loaderId === undefined || this.#currentNavigation === navigation) {
-      // If the command's navigation is same-document or is already the current one,
-      // nothing to do.
-      return;
-    }
-
-    this.#currentNavigation.fail(
-      'navigation canceled by concurrent navigation',
-    );
-
-    navigation.start();
-    this.#currentNavigation = navigation;
+  frameStartedNavigating(
+    url: string,
+    loaderId: string,
+    navigationType: string,
+  ) {
+    this.#logger?.(LogType.debug, `frameStartedNavigating ${url}, ${loaderId}`);
 
-    if (this.#pendingNavigation === navigation) {
+    if (
+      this.#pendingNavigation &&
+      this.#pendingNavigation?.loaderId !== undefined &&
```

Comment:
Nit: You can omit the optional chain for `this.#pendingNavigation`

Reply:
correct, but I find it slightly better for readability